documents-services/src/main/java/org/exoplatform/documents/rest/DocumentFileRest.java

```diff
@@ -1298,5 +1302,58 @@ public Response importDocuments(@Parameter(description = "upload id", required =
     }
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("administrators")
+  @Path("/deleted")
+  @Operation(summary = "Fetch trash documents", method = "GET", description = "Fetch trash document")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "400", description = "Invalid query input"),
+      @ApiResponse(responseCode = "401", description = "Unauthorized operation"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"), })
+  public Response getDeletedDocuments(@QueryParam("sortField")
+                                      String sortField,
+                                      @Parameter(description = "Sort ascending or descending")
+                                      @QueryParam("sortDirection")
+                                      String sortDirection,
+                                      @Parameter(description = "Offset of results to return")
+                                      @QueryParam("offset")
+                                      int offset,
+                                      @Parameter(description = "Limit of results to return")
+                                      @Schema(defaultValue = "20")
+                                      @QueryParam("limit")
+                                      int limit) {
+    if (limit < 0 || offset < 0) {
+      LOG.error("Invalid offset or limit parameters");
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    TrashElementNodeFilter filter = new TrashElementNodeFilter();
+    if (StringUtils.isBlank(sortField)) {
+      sortField = "modifiedDate";
+    }
+    if (StringUtils.isBlank(sortDirection)) {
+      sortDirection = "desc";
+    }
+    filter.setAscending(sortDirection.equalsIgnoreCase("asc"));
+    filter.setOffset(offset);
+    filter.setLimit(limit);
+    filter.setSortField(getFromAlias(sortField));
+    try {
+      List<TrashElementNode> trashElementNodes = documentFileService.getDeletedDocuments(filter);
+      int size = documentFileService.countDeletedDocuments();
+      List<TrashElementEntity> trashElementEntities = trashElementNodes.stream()
+                                                                       .map(EntityBuilder::toTrashElement)
+                                                                       .collect(Collectors.toList());
+      CollectionEntity<TrashElementEntity> collectionEntity = new CollectionEntity<>(trashElementEntities,
```

Comment:
This is not performant, as you get all trash nodes, then you apply limit and offset AFTER. 
We need to pass offset and limit to documentFileService, which take care of offset and limit to return the correct list.

Reply:
Limit and offset have already been passed to the service and applied to the query. This stream operation is only used to convert the returned list from `ElementTrashNode` to the DTO `ElementTrashEntity`!

Reply:
does both stream can be combined ?
Why do we use 2 ?

Reply:
where is the second stream! , we have only one stream , the second one is a constructor of the CollectionEntity DTO ?!!

Reply:
pff, sorry, I read to quickly ...